Source/Core/Core/PowerPC/JitInterface.h

```diff
@@ -71,6 +63,20 @@ class JitInterface
   // the JIT'ed code.
   void ClearSafe();
 
+  // DolphinQt's JITWidget needs EraseSingleBlock. Nothing else (from outside of the Core) should
+  // use it, or else JitBlockTableModel will contain a dangling reference. If something else from
+  // outside of the Core *must* use this, consider reworking the logic in JITWidget.
+  void EraseSingleBlock(const JitBlock& block);
+
+  using MemoryStats = std::pair<std::string_view, std::pair<std::size_t, double>>;
```

Comment:
It currently is a `std::pair<std::size_t, double>` because the type originates from the rangeset external maintained by AdmiralCurtiss. Granted, I chose the type when I updated the upstream with my changes, but it was simplest this way; the type needed to exist in both rangeset.h and rangesizeset.h, and without using std::pair that would require a third header just to define a pair-like type.
I know, excuses, but those are the reasons for what I went with here. I can add a comment in JitBase.h mirroring the one in rangeset.h/rangesizeset.h if that helps? I will be using another structured binding where this pair is ultimately used in JitWidget.cpp to self-document the meaning of the size_t and double pair in that location.

Reply:
How about something like:

```
using FreeSize = std::size_t;
using FragmentationRatio = double;
using MemoryStats = std::pair<std::string_view, std::pair<FreeSize, FragmentationRatio>>;
```

Reply:
I understand the approach—documentation in type names doesn't hold the same risk of becoming outdated like comment documentation does—but defining types that aren't useful anywhere seems silly. Are the comments not enough?

Reply:
Well, I guess comments also work. Not that I see why they would be better.